File: scatter_plot.R
# R script to create a scatter plot and save it with a random filename

# Generate random data for the scatter plot
set.seed(123)  # For reproducible results
x <- rnorm(100)
y <- 2 * x + rnorm(100, sd = 0.5)

# Generate a random filename
random_string <- paste(sample(c(letters, LETTERS, 0:9), 10, replace = TRUE), collapse = "")
filename <- paste0("scatter_plot_", random_string, ".png")

# Create the scatter plot
plot(x, y, 
     main = "Scatter Plot", 
     xlab = "X Values", 
     ylab = "Y Values",
     pch = 19,  # solid circles
     col = "blue")

# Add a trend line
abline(lm(y ~ x), col = "red", lwd = 2)

# Save the plot with the random filename
png(filename, width = 800, height = 600)
plot(x, y, 
     main = "Scatter Plot", 
     xlab = "X Values", 
     ylab = "Y Values",
     pch = 19,  # solid circles
     col = "blue")
abline(lm(y ~ x), col = "red", lwd = 2)
dev.off()

cat("Scatter plot saved as:", filename, "\n")

File: scatter_plot.py
import matplotlib.pyplot as plt
import numpy as np
import random
import string

# Generate random filename
random_name = 'scatter_plot_' + ''.join(random.choices(string.ascii_lowercase + string.digits, k=8)) + '.png'

# Create sample data
x = np.random.randn(100)
y = 2 * x + np.random.randn(100)

# Create the scatter plot
plt.figure(figsize=(10, 6))
plt.scatter(x, y, alpha=0.6, color='blue')
plt.title('Scatter Plot: X vs Y')
plt.xlabel('X values')
plt.ylabel('Y values')

# Add a trend line
z = np.polyfit(x, y, 1)
p = np.poly1d(z)
plt.plot(x, p(x), "r--", alpha=0.8, label='Trend line')
plt.legend()

# Save the plot with the random filename
plt.savefig(random_name, dpi=300, bbox_inches='tight')
plt.show()

print(f"Scatter plot saved as: {random_name}")